h/static/scripts/widget-controller.js

```diff
@@ -13,28 +28,46 @@ module.exports = function WidgetController(
   $scope.threadRoot = threading.root;
   $scope.sortOptions = ['Newest', 'Oldest', 'Location'];
 
-  var _resetAnnotations = function () {
+  function _resetAnnotations() {
     // Unload all the annotations
     annotationMapper.unloadAnnotations(threading.annotationList());
     // Reload all the drafts
     threading.thread(drafts.unsaved());
-  };
+  }
 
   var searchClients = [];
 
   function _loadAnnotationsFor(uri, group) {
-    var searchClient = new SearchClient(store.SearchResource);
+    var searchClient = new SearchClient(store.SearchResource, {
+      // If no group is specified, we are fetching annotations from
+      // all groups in order to find out which group contains the selected
+      // annotation, therefore we need to load all chunks before processing
+      // the results
```

Comment:
This is fine. What I supposed I had actually envisioned was that we would always fetch all annotations, and wouldn't then need to issue an additional query when the user switched between groups. I recognise that's probably a lot more work (and would need changes to the way the streamer works, too) but it's worth bearing in mind as a possible simplification in the future.